### ISSUE TITLE
Advanced Preferences can't handle "24:00"

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#340]
**Reported by Didactylos on 16 Sep 37574525 05:46 UTC**
Advanced Preferences can't handle the value "24:00". Neither can the simple prefs, but that doesn't matter since "Never" isn't a useful value unless you can override it with the per-day settings.

(This is almost-but-not-quite a duplicate of #300)

But this is just another bug in the advanced preferences. If it were my call, I'd bump the entire advanced preferences dialog to 6.0.

Also see #351, #352 and #173.

**Comment by romw** [comment]
**Commented by Nicolas on 4 Sep 37574568 16:26 UTC**
,,Just a sidenote: you can link to the [advanced_preferences tag](query:keywords~=advanced_preferences) to see all the related tickets.,,

**Comment by romw** [comment]
**Commented by Ageless on 11 Sep 37575524 22:13 UTC**
Replying to [Didactylos](ticket:357):
> Advanced Preferences can't handle the value "24:00".

If you mean in the time slots, then that's normal, isn't it? Digital clocks don't show that it is 24:00 either at midnight. It changes from 23:59 to 0:00.

So the entries go from 0:00 to 23:59.

**Comment by romw** [comment]
**Commented by Didactylos on 4 Apr 37576242 06:13 UTC**
Replying to [Ageless](comment:2):
Almost correct. Midnight is 00:00 ''and'' 24:00 - although since clocks can't display two values at once, [nearly](http://en.wikipedia.org/wiki/Image:Clock_showing_24_00.JPG) all of them show 00:00. But BOINC treats 24:00 as exactly what it represents; midnight at the end of the day. As you can see, using this value is the only way to specify that BOINC ''never'' compute or connect. Here's the actual code:

```
    if (start==end) return false;
    if (start==0 && end==24) return false;
    if (start==24 && end==0) return true;
    if (start < end) {
        return (hour < start || hour > end);
    } else {
        return (hour >= end && hour < start);
    }
```

**Comment by romw** [comment]
**Commented by Didactylos on 2 Feb 37747758 22:40 UTC**
I think this is bumped to milestone:6.0 now.

**Comment by romw** [comment]
**Commented by Nicolas on 15 Feb 37760321 04:26 UTC**
Bumped.

**Comment by romw** [comment]
**Commented by Didactylos on 26 Sep 37879677 23:33 UTC**
Combined #300 with this ticket.

**Comment by romw** [comment]
**Commented by davea on 29 Apr 39632927 13:20 UTC**
(In [18866]) - manager: allow 24:00 to be used in advanced prefs dialog
    (from Ian Hay; fixes #357)

**Comment by romw** [comment]
**Commented by romw on 27 Dec 39651733 14:40 UTC**
(In [18909])     - manager: allow 24:00 to be used in advanced prefs dialog
        (from Ian Hay; fixes #357)

    clientgui/
        DlgAdvPreferences.cpp
